### PR TITLE
cleanup unix sockets

### LINKED ISSUE #10969: Delete unix sockets on daemon shutdown
**Summary**: 

The daemon used to clean up its unix sockets after #6962 got merged, but there was a regression in #7317 and now it's not the case anymore.

**Proposed solution**:
Implement `removeUnixSockets` as a method on `Daemon` and call it during daemon shutdown.

### PR DESCRIPTION
**Pull request by cpuguy83**
Closes #10969

Carry of #11085

### CODE REVIEW

File: engine/engine.go [hunk 1]
```diff
@@ -156,8 +157,11 @@ func (eng *Engine) Shutdown() {
 	eng.l.Lock()
 	if eng.shutdown {
 		eng.l.Unlock()
+		eng.shutdownLock.Lock()
+		defer eng.shutdownLock.Unlock()
```
Comment (estesp) [inline engine/engine.go]: Trying to get my head around this one! :) your new lock is locked, and then immediately unlocked with the return on 162?

**Comment by cpuguy83** [comment]
Accidently forced pushed while the old PR was closed, #11476 for reference.

**Comment by jfrazelle** [comment]
LGTM, @estesp wpuld you mind looking at the engine.Shutdown since you just changed something with that too :)

File: engine/engine.go [hunk 2]
```diff
@@ -180,17 +183,15 @@ func (eng *Engine) Shutdown() {
...
+			eng.shutdownWait.Done()
 		}(h)
 	}
 	done := make(chan struct{})
 	go func() {
-		wg.Wait()
+		eng.shutdownWait.Wait()
 		close(done)
 	}()
 	select {
```
Comment (estesp) [inline engine/engine.go]: now that we have the `shutdownWait` group, and we are waiting on it in the goroutine, do we still want this other channel with 10s timeout? I mean, behavior-wise I guess it is exactly the same for this sequence of actions, as the `eng.shutdownWait` replaces `wg`; but in the broader sense, the couple of PRs here have tried to make shutdown much more sane/controlled, and I wonder if that timeout is too short for the case of a significantly loaded system, maybe running devmapper, and doing a lot of cleanup?

Comment (cpuguy83) [inline engine/engine.go]: I agree, I don't like the timeout... but probably a broader discussion is needed here.

Comment (estesp) [inline engine/engine.go]: cool, let's let that be a separate discussion then

**Comment by estesp** [comment]
LGTM

**Pull request merged**